cunumeric/array.py

```diff
@@ -955,12 +955,101 @@ def byteswap(self, inplace=False):
             numpy_array = self.__array__(stacklevel=3).byteswap(inplace=False)
             return self.convert_to_cunumeric_ndarray(numpy_array, stacklevel=3)
 
-    @unimplemented
-    def choose(self, choices, out, mode="raise"):
-        numpy_array = self.__array__(stacklevel=3).choose(
-            choices=choices, out=out, mode=mode
-        )
-        return self.convert_to_cunumeric_ndarray(numpy_array, stacklevel=3)
+    def choose(self, choices, out=None, mode="raise", stacklevel=2):
+        a = self.convert_to_cunumeric_ndarray(self)
+        if out is not None:
+            out = out.convert_to_cunumeric_ndarray(out)
+        # choices can be passed as array or tuple of arrays.
+        # in case of the tuple, we will pass each array separately
+        is_tuple = isinstance(choices, tuple)
```

Comment:
Based on the NumPy documentation for choose*, we should support both lists and tuples here. You may want to just fall back to the tuple case:
```
if isinstance(choices, list):
  choices = tuple(choices)
```
Currently lists work by first converting to a cunumeric ndarray (the function `convert_to_cunumeric_ndarray` accepts many kinds of containers), which is not optimal.

\* relevant bit from https://numpy.org/doc/stable/reference/generated/numpy.choose.html: "the outermost sequence-like container should be either a list or a tuple."